app.js

```diff
@@ -1,162 +1,241 @@
+  "Need": 1,
+  "Feature": 2,
+  "Use Case": 3,
+  "User Story": 4,
+  "Quality": 5,
+  "Design Element": 6,
+};
+
+var requirementObj = {
   "RequirementId": null,
-  "Name": "test",
-  "Description": "test Description",
-  "ReleaseVersionNumber": null,
-  "RequirementTypeName": null,
-  "ImportanceName": null,
-  "StatusName": null,
+  "Name": null,
+  "Description": null,
+  "ReleaseId": null,
+  "RequirementTypeId": null,
+  "ImportanceId": null,
+  "StatusId": null,
   "EstimatePoints": null,
-  "AuthorName": null,
-  "OwnerName": null,
-  "ComponentName": null
+  "AuthorId": null,
+  "OwnerId": null,
+  "ComponentId": null,
 };
 
-(function () {
+var reqStatus = {
+  "Requested":	1,
+  "Evaluated":	7,
+  "Accepted":	5,
+  "Rejected":	6,
+  "Planned":	2,
+  "In Progress":	3,
+  "Developed":	4,
+  "Obsolete":	8,
+  "Tested":	9,
+  "Completed":	10,
+}
+
+//column ranges for different sheets, currently only requirements.
+var columnRanges = {
+  requirements: "A3:K"
+};
 
+//custom field ranges for different sheets, currently only requirements.
+var customFieldRanges = {
+  requirements: ["N", "AQ"],
+};
 
-  // The initialize function must be run each time a new page is loaded
-  Office.initialize = function (reason) {
-    $(document).ready(function () {
-      $('#logIn').click(logIn);
-      $('#testing').click(testing);
-      $('#help-toggle').click(showHelp);
-      $('#export').click(getRowAmount);
-    })
-  };
+//array to hold custom field names
+var customFieldNames = [];
+
+function cleanObject(Obj) {
+  var cleaned = {};
+  for (let prop in Obj) {
+    if (Obj[prop] != "") {
+      cleaned[prop] = Obj[prop];
+    }
+  }
+  return cleaned;
+}
+
+//converts from Excel days since 1/1/1990 to Spira milliseconds since 1/1/1970
+function daysToMseconds(days) {
+  days -= 2; //for some reason excel returns 2 extra days?
+  const between = 25566;
+  days -= between;
+  let milliseconds = days *= 8.64e+7;
+  return `/Date(${milliseconds})/`;
+}
+
+function disableButtons() {
+  $('button').attr('disabled', 'disabled');
+  $('button').addClass('is-disabled');
+  $('select').attr('disabled', 'disabled');
+  $('select').addClass('is-disabled');
+}
+
+function enableButtons() {
+  $('button').removeClass('is-disabled');
+  $('button').prop('disabled', false);
+  $('select').removeClass('is-disabled');
+  $('select').prop('disabled', false);
+}
+
+function multilistConvert(str) {
+    console.log(str);
+    function correct(str) {
+      let newString = parseInt(str.trim());
+      return newString;
+    }
+    let arr = str.split(",");
+    arr = arr.map(correct);
+    arr = arr.filter(Number.isInteger);
+    return arr;
+  }
+
+function toIdString(artifact) {
+  let newString = toSheetName(artifact);
+  newString = newString.split(" ");
+  newString = newString.join("");
+  newString = newString.split("");
+  newString[newString.length - 1] = "Id";
+  newString = newString.join("");
+  return newString;
+}
+
+function toSheetName(artifact) {
```

Comment:
changed toSheetName and toIdString into convertToSheetName(artifactName) and convertToIdKey(artifactName), and added comment, hope it helps